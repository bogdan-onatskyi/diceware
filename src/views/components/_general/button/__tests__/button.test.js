import React from 'react';

import Button from '../button';

describe("Button component", () => {
    it("should match saved snapshot (type === 'sometype')", () => {
        const handleClick = jest.fn();
        const wrapper = shallow(
            <Button type="sometype" data-tip="Tooltip" onClick={handleClick}>
                Button text
            </Button>
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.simulate('click');
        expect(handleClick).toHaveBeenCalled();
    });

    it("should match saved snapshot (disabled === true)", () => {
        const handleClick = jest.fn();
        const wrapper = shallow(
            <Button type="sometype" disabled={true} selected={false}
                    data-tip="Tooltip" onClick={handleClick}>
                Button text
            </Button>
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.simulate('click');
        expect(handleClick).not.toHaveBeenCalled();
    });

    it("should match saved snapshot (selected === true)", () => {
        const wrapper = shallow(
            <Button type="sometype" disabled={false} selected={true}
                    data-tip="Tooltip">
                Button text
            </Button>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should match saved snapshot (data-tip === '')", () => {
        const wrapper = shallow(
            <Button type="sometype" disabled={false} selected={false}
                    data-tip="">
                Button text
            </Button>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should have 4 props", () => {
        const propsStr = 'className type selected disabled onClick';
        const props = [];
        for (let prop in Button.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});
