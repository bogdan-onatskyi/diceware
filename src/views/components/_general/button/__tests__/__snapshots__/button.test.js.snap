// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match saved snapshot (data-tip === '') 1`] = `
<button
  className="button button--sometype"
  data-tip=""
>
  Button text
</button>
`;

exports[`Button component should match saved snapshot (disabled === true) 1`] = `
<button
  className="button button--disabled button--sometype"
  data-tip="Tooltip"
  onClick={null}
>
  Button text
</button>
`;

exports[`Button component should match saved snapshot (selected === true) 1`] = `
<button
  className="button button--selected button--sometype"
  data-tip="Tooltip"
>
  Button text
</button>
`;

exports[`Button component should match saved snapshot (type === 'sometype') 1`] = `
<button
  className="button button--sometype"
  data-tip="Tooltip"
  onClick={[Function]}
>
  Button text
</button>
`;
